Created adapter for timedelta and added tests related to new adapter

AlexTorx opened the pull request:
I noticed there was no adapter for datetime.timedelta json rendering object.

I created one using the timedelta.total_seconds method from datetime.timedelta object as a serialization method.

jssuzanne commented:
great. Why second only ? Maibe a parameter to select between second, micro-second or other unit should be fun if we want other than second

AlexTorx commented:
I have added a mode parameter to select how timedelta objects should be serialized (microseconds, milliseconds, seconds, minutes, hours, days or weeks).

Tests now cover these changes.

Updated copyrights

jssuzanne commented on anyblok_pyramid/adapter.py:
```diff
@@ -49,10 +49,66 @@ def timedelta_adapter(obj, request):
+        * minutes
+        * hours
+        * days
+        * weeks
```
You should create constante or enum to help

jssuzanne commented on anyblok_pyramid/tests/test_adapter.py:
```diff
@@ -39,6 +42,59 @@ def add_route_and_views(config):
+            json_renderer.add_adapter(
+                timedelta, lambda obj, request: timedelta_adapter(
+                    obj, request, mode))
+            config.add_renderer('json', json_renderer)
```
it's complicated to understand the use with a lamda.

Maybe the better action is to decorate the timedelta_adapter

```
def timedelta_factory(mode=SECONDS):
    def timedelta_adapter(obj, request):
          # copy your adapter here

    return wrap
```

so the configuration is easer

```
from anyblok_pyramid.adapter import timedelta_factory, MILLISECONDS

...

json_renderer.add_adapter(timedelta, timedelta_factory(MILLISECONDS))
```

jssuzanne commented:
> I have added a mode parameter to select how timedelta objects should be serialized (microseconds, milliseconds, seconds, minutes, hours, days or weeks).
> 
> Tests now cover these changes.
> 
> Updated copyrights

You miss to add notification in the change log

jssuzanne commented on anyblok_pyramid/adapter.py:
```diff
@@ -40,65 +42,88 @@ def datetime_adapter(obj, request):
+        :rtype: str, seconds corresponding to timedelta
+        """
+
+        import ipdb; ipdb.set_trace()
```
Oups !!!

AlexTorx commented:
> Great, thank for this PR, I let you merge it

Yeah that would be really cool, but I guess I do not have any right in order to perform that...
